src/diffusers/loaders/single_file_model.py

```diff
@@ -67,6 +67,23 @@
 }
 
 
+def _is_subclass(src_cls, dst_cls_str):
+    """
+    Find if src_cls is a subclass of dst_cls whose name is dst_cls_str
+    """
+    for cls in src_cls.__mro__:
+        if cls.__name__ == dst_cls_str:
+            return True
+    return False
+
+
+def _get_single_file_loadable_mapping_class(cls):
```

Comment:
Nice! I think we can leverage Python's built in `issubclass` here if we use 

```suggestion
def _get_single_file_loadable_mapping_class(cls):
    diffusers_module = importlib.import_module(__name__.split(".")[0])
    for loadable_class_str in SINGLE_FILE_LOADABLE_CLASSES:
        loadable_class = getattr(diffusers_module, loadable_class_str)

        if issubclass(cls, loadable_class):
            return loadable_class_str

    return None
```

Then we just need a single function to do the check.

Reply:
Sure! This is a more elegant approach indeed.